### PR TITLE
Create shared formatToProposal util

### PR DESCRIPTION
**Pull request by therealemjy**
## Changes

- create `formatToProposal` to prevent duplicated code

### CODE REVIEW

File: src/utilities/formatToProposal.ts [hunk 1]
```diff
@@ -51,22 +87,13 @@ const formatToProposal = ({
     descriptionObj = { version: 'v1' as const, title: plainTitle, description: descriptionText };
   }
 
-  if (!endDate) {
```
Comment (coreyar) [inline src/utilities/formatToProposal.ts:54]: Are we guaranteed to have an endDate? I took this from the original implementation that was calculate the end date based on the endBlock if it was missing

File: src/utilities/formatToProposal.ts [hunk 2]
```diff
@@ -87,6 +114,16 @@ const formatToProposal = ({
     startTxHash: startTxHash ?? undefined,
     totalVotesWei: abstainedVotesWei.plus(againstVotesWei).plus(forVotesWei),
   };
+
+  if (actions) {
```
Comment (coreyar) [inline src/utilities/formatToProposal.ts]: Actions are required to submit a proposal so there should always be at least 1

File: src/utilities/formatToProposal.ts (hunk 1 above)
Comment (therealemjy) [reply src/utilities/formatToProposal.ts:54]: Nope, that's a mistake on my end! Good catch

File: src/pages/Proposal/index.tsx [hunk 3]
```diff
@@ -105,7 +105,7 @@ export const ProposalUi: React.FC<ProposalUiProps> = ({
         />
       </div>
 
-      <Description description={proposal.description} actions={proposal.actions} />
+      <Description description={proposal.description} actions={proposal.actions || []} />
```
Comment (coreyar) [inline src/pages/Proposal/index.tsx]: An alternative philosophy is to format all data in the `format` functions so it is usable in the frontend. For example, arrays returned as undefined would be set as an empty array when formatting.

File: src/utilities/formatToProposal.ts (hunk 2 above)
Comment (therealemjy) [reply src/utilities/formatToProposal.ts]: They are yes, but the `/voters/history/` endpoint does not return any `actions` field which is why this check is here

Comment (coreyar) [reply src/utilities/formatToProposal.ts]: Interesting, that seems like a bug on the endpoint

Comment (therealemjy) [reply src/utilities/formatToProposal.ts]: My thought as well. In the meantime, I've followed your advice of assigning an empty array of no action was present in the input

**Review by coreyar — APPROVED** [review]

**Pull request merged**